### ISSUE TITLE
[Accessibility] Rating in addition with the product name fails to announce after focusing on the "Product" title for "All reviews" block

### ISSUE THREAD

**Issue by mdperez86** [woothemes/woocommerce#56224]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

Rating in addition with the product name fails to announce after focusing on the "Product" title for "All reviews" block. This was tested using MacOS BigSur 11.7.10, Safari and Chrome and it is a confirmed issue. It's also important to consider that this is currently working in MacOS Sequoia 15.3.1, Safari and Chrome. So the fix and test must be made considering this.

The issue was introduced in https://github.com/woocommerce/woocommerce/pull/54811 when working to fix https://github.com/woocommerce/woocommerce/issues/52133

The issue
![Image](https://github.com/user-attachments/assets/03e61d2a-7913-4e2c-af9e-d23128fc699d)

It should be announced this way
![Image](https://github.com/user-attachments/assets/d124a69d-aba0-4e0a-8408-aa079834821f)

### Component or Page: 

Component: All Reviews Block

### Possible cause:

The problem could be related to the [`aria-hidden="true"`](https://github.com/woocommerce/woocommerce/pull/54811/files#diff-d5480bbc85a67ebe7e6587f2e3b8c996e8fd764f6567e9c9578755e7263f4310R140) that could conflict with the [`aria-describedby`](https://github.com/woocommerce/woocommerce/pull/54811/files#diff-d5480bbc85a67ebe7e6587f2e3b8c996e8fd764f6567e9c9578755e7263f4310R86) in older versions of the MacOS/Safari/VO

**Comment by vikrampm1** [comment]
@dinhtungdu please check if this relates to something you've worked on recently or have our team look into it.

**Comment by dinhtungdu** [comment]
@vikrampm1 I haven't worked on this block, feel free to take it, thanks!
